### PR TITLE
Sp6 388 522 contacts notification fe

### PR DESCRIPTION
**Pull request by serhiibets**

### CODE REVIEW

File: wayat/frontend/lib/app_state/notification_state/notification_state.dart [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+    await flutterLocalNotificationsPlugin
+        .resolvePlatformSpecificImplementation<
+            AndroidFlutterLocalNotificationsPlugin>()
+        ?.createNotificationChannel(channel);
+  }
+
+  /// Set token if refreshed
+  void setToken(String token) {
+    NotificationServiceImpl().sendCurrentUserToken(token);
+    print(
```
Comment (capf4bbo) [inline wayat/frontend/lib/app_state/notification_state/notification_state.dart:166]: Please remove all the prints

File: wayat/frontend/lib/app_state/notification_state/notification_state.dart [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+          '-----------------------------------------Send token to backend: $token');
+    });
+    _tokenStream = messagingInstance.onTokenRefresh;
+    _tokenStream.listen(setToken);
+  }
+
+  /// Return notification with [newMessage]
+  PushNotification pushNotification(RemoteMessage newMessage) {
+    if (platformService.isWeb) {
+      // TODO: implement web notification
```
Comment (capf4bbo) [inline wayat/frontend/lib/app_state/notification_state/notification_state.dart:184]: Please remove this if

File: wayat/frontend/lib/app_state/notification_state/notification_state.dart [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+      trailing: OutlinedButton(
+        style: OutlinedButton.styleFrom(
+          backgroundColor: Colors.white,
+          foregroundColor: Colors.black,
+          shape: RoundedRectangleBorder(
+            borderRadius: BorderRadius.circular(1000.0),
+            side: const BorderSide(color: Colors.black, width: 1),
+          ),
+        ),
+        onPressed: (() {
```
Comment (capf4bbo) [inline wayat/frontend/lib/app_state/notification_state/notification_state.dart:253]: If this is the dismiss button please remove it, if not remove the TODO comment

**Review by capf4bbo — CHANGES_REQUESTED** [review]

File: wayat/frontend/lib/features/notification/widgets/dynamic_dialog.dart [hunk 4]
```diff
@@ -0,0 +1,31 @@
+// ignore_for_file: public_member_api_docs, sort_constructors_first
+//push notification dialog for foreground
+import 'package:flutter/material.dart';
+
```
[the remark this answers is not in the document]
Comment (serhiibets) [reply wayat/frontend/lib/features/notification/widgets/dynamic_dialog.dart:4]: It's unused file. I delete it from project.

File: wayat/frontend/lib/features/notification/widgets/notification_badge.dart [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import 'package:flutter/material.dart';
+import 'package:wayat/common/theme/colors.dart';
+
```
[the remark this answers is not in the document]
Comment (serhiibets) [reply wayat/frontend/lib/features/notification/widgets/notification_badge.dart:3]: Done

File: wayat/frontend/lib/services/notification/mock/notification_service.dart [hunk 6]
```diff
@@ -0,0 +1,3 @@
+abstract class NotificationService {
```
[the remark this answers is not in the document]
Comment (serhiibets) [reply wayat/frontend/lib/services/notification/mock/notification_service.dart:1]: Done

File: wayat/frontend/lib/services/notification/mock/notification_service_impl.dart [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import 'package:get_it/get_it.dart';
```
[the remark this answers is not in the document]
Comment (serhiibets) [reply wayat/frontend/lib/services/notification/mock/notification_service_impl.dart:1]: Done

File: wayat/frontend/lib/features/notification/widgets/dynamic_dialog.dart [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+class DynamicDialog extends StatefulWidget {
```
[the remark this answers is not in the document]
Comment (serhiibets) [reply wayat/frontend/lib/features/notification/widgets/dynamic_dialog.dart:5]: It's unused file. I delete it.

File: wayat/frontend/lib/app_state/notification_state/notification_state.dart (hunk 1 above)
Comment (serhiibets) [reply wayat/frontend/lib/app_state/notification_state/notification_state.dart]: done

File: wayat/frontend/lib/app_state/notification_state/notification_state.dart (hunk 2 above)
Comment (serhiibets) [reply wayat/frontend/lib/app_state/notification_state/notification_state.dart:177]: Done

File: wayat/frontend/lib/app_state/notification_state/notification_state.dart (hunk 3 above)
Comment (serhiibets) [reply wayat/frontend/lib/app_state/notification_state/notification_state.dart]: Done

**Review by josuecapgemini — APPROVED** [review]

**Review by josuecapgemini — APPROVED** [review]

**Review by capf4bbo — APPROVED** [review]

**Review by econesag — APPROVED** [review]

**Review by josuecapgemini — APPROVED** [review]

**Pull request merged**